pkg/apis/planetscale/v2/vitessbackupschedule_types.go

```diff
@@ -0,0 +1,217 @@
+	// runs until Suspend is set to false again. This is useful if you want to pause backup without
+	// having to remove the entire VitessBackupSchedule object from the cluster.
+	// +optional
+	Suspend *bool `json:"suspend,omitempty"`
+
+	// StartingDeadlineSeconds enables the VitessBackupSchedule to start a job even though it is late by
+	// the given amount of seconds. Let's say for some reason the controller process a schedule run on
+	// second after its scheduled time, if StartingDeadlineSeconds is set to 0, the job will be skipped
+	// as it's too late, but on the other hand, if StartingDeadlineSeconds is greater than one second,
+	// the job will be processed as usual.
+	// +optional
+	// +kubebuilder:validation:Minimum=0
+	StartingDeadlineSeconds *int64 `json:"startingDeadlineSeconds,omitempty"`
+
+	// ConcurrencyPolicy specifies ho to treat concurrent executions of a Job.
+	// Valid values are:
+	// - "Allow" (default): allows CronJobs to run concurrently;
+	// - "Forbid": forbids concurrent runs, skipping next run if previous run hasn't finished yet;
+	// - "Replace": cancels currently running job and replaces it with a new one.
+	// +optional
+	// +kubebuilder:example="Forbid"
+	ConcurrencyPolicy ConcurrencyPolicy `json:"concurrencyPolicy,omitempty"`
+
+	// AllowedMissedRuns defines how many missed run of the schedule will be allowed before giving up on finding the last job.
+	// If the operator's clock is skewed and we end-up missing a certain number of jobs, finding the last
+	// job might be very time-consuming, depending on the frequency of the schedule and the duration during which
+	// the operator's clock was misbehaving. Also depending on how laggy the clock is, we can end-up with thousands
+	// of missed runs. For this reason, AllowedMissedRun, which is set to 100 by default, will short circuit the search
+	// and simply wait for the next job on the schedule.
+	// Unless you are experiencing issue with missed runs due to a misconfiguration of the clock, we recommend leaving
+	// this field to its default value.
+	// +optional
+	// +kubebuilder:validation:Minimum=0
+	AllowedMissedRuns *int `json:"allowedMissedRun,omitempty"`
+
+	// JobTimeoutMinutes defines after how many minutes a job that has not yet finished should be stopped and removed.
+	// Default value is 10 minutes.
+	// +optional
+	// +kubebuilder:validation:Minimum=0
+	// +kubebuilder:default=10
+	JobTimeoutMinutes int32 `json:"jobTimeoutMinute,omitempty"`
+}
+
+// VitessBackupScheduleStrategy defines how we are going to take a backup.
+// The VitessBackupSchedule controller uses this data to build the vtctldclient
+// command line that will be executed in the Job's pod.
+type VitessBackupScheduleStrategy struct {
+	// Name of the backup strategy.
+	Name BackupStrategyName `json:"name"`
+
+	// Cluster defines on which cluster you want to take the backup.
+	// This field is mandatory regardless of the chosen strategy.
+	Cluster string `json:"cluster"`
```

Comment:
i'm not sure i follow why this is necessary. my mental model is that a user defines `[]VitessBackupScheduleTemplate` on the `ClusterBackupSpec`, and so implicitly each `VitessBackupScheduleStrategy` will be associated with the cluster where `ClusterBackupSpec` is defined.

Reply:
That's a good point @maxenglander, it is pretty useless. I ended up removing that field from `VitessBackupScheduleStrategy` and adding it to `VitessBackupScheduleSpec`. The `VitessCluster` controller will come and fill that new field when it create a new `VitessBackupSchedule` object, that way `VitessBackupSchedule` is still be able to select existing components given their cluster names to avoid fetching wrong data in the event where we have multiple `VitessCluster` running in our K8S cluster.

See b30aa09 for the change.